pennylane/labs/intermediate_reps/rowcol.py

```diff
@@ -0,0 +1,617 @@
+
+from typing import Iterable
+
+import galois
+import networkx as nx
+import numpy as np
+from networkx.algorithms.approximation import steiner_tree
+
+
+def postorder_traverse(tree: nx.Graph, source: int, source_parent: int = None):
+    """Post-order traverse a tree graph, starting from (but excluding) the node ``source``.
+
+    Args:
+        tree (nx.Graph): Tree graph to traverse. Must contain ``source``. Must contain
+            ``source_parent`` if it is not None. Typing assumes integer-labeled nodes.
+        source (int): Node to start the traversal from
+        source_parent (Optional[int]): Parent node of ``source`` in ``tree``.
+
+    Returns:
+        list[tuple[int]]: Pairs of nodes that constitute post-order traversal of ``tree``
+        starting at ``source``. Strictly speaking, the traversal is encoded in the first
+        entry of each pair, and the second entry simply is the parent node for each first entry.
+
+    A useful illustration of depth-first tree traversals can be found on
+    `Wikipedia <https://en.wikipedia.org/wiki/Tree_traversal#Depth-first%20search>`__.
+
+    **Example**
+
+    Consider the tree
+
+    ```
+    #                   (4)
+    #                    |
+    # (6) - (2) - (0) - (1) - (3) - (8)
+    #        |           |
+    #       (7)         (5)
+    ```
+
+    and consider ``(0)`` to be the source, or root, of the tree.
+    We may construct this tree as a ``nx.Graph`` by providing the edge data:
+
+    >>> import networkx as nx
+    >>> G = nx.Graph([(0, 1), (0, 2), (1, 3), (1, 4), (1, 5), (2, 6), (2, 7), (3, 8)])
+
+    As for every tree traversal, post-order traversal results in a reordering of the nodes of
+    the tree, with each node appearing exactly once. Post-order traversing the graph means that
+    for every node we reach, we first visit its child nodes (in standard sorting of the node
+    labels/indices) and then append the node itself to the ordering.
+    Starting at the root ``(0)``, we thus perform the following steps:
+
+    1. Visit ``(1)``, ``(3)`` and then ``(8)``, without appending any of them.
+    2. Append ``(8)`` because there are no child nodes to visit (it is a leaf node).
+    3. Append ``(3)`` because all children of it have been visited.
+    4. Visit the next child of ``(1)``, which is ``(4)``, and append it because it is a leaf node.
+    5. Visit the last child of ``(1)``, which is ``(5)``, and append it because it is a leaf node.
+    6. Append ``(1)`` because all its children have been visited.
+    7. Visit ``(2)`` and then ``(6)``, the latter of which is appended (leaf node).
+    8. Visit the second and last child of ``(2))``, which is ``(7)``, and append it (leaf node).
+    9. Append ``(2)`` becaues all its children have been visited.
+    10. Append ``(0)`` becaues all its children have been visited.
+
+    Overall, the post-order traversal reads ``[8, 3, 4, 5, 1, 6, 7, 2, 0]``.
```

Comment:
Could you put this before? I think it is better to see the result first, and then  explain why